Source/DataSources/RectangleGeometryUpdater.js

```diff
@@ -453,8 +471,16 @@ define([
         var outlineWidth = rectangle.outlineWidth;
         var closeBottom = rectangle.closeBottom;
         var closeTop = rectangle.closeTop;
+        var onTerrain = fillEnabled && !defined(height) && !defined(extrudedHeight) &&
+                        isColorMaterial && GroundPrimitive.isSupported(this._scene);
+
+        if (outlineEnabled && onTerrain) {
+            oneTimeWarning('Entities with an outline are unsupported on terrain. Outline will be disabled.');
```

Comment:
Since this message is copied in 4 places, I would add a simple property to `oneTimeWarning` like `oneTimeWarning.geometryOutlines` with this string and use that instead.

Also, we should replace this message with `Entity geometry outlines are unsupported on terrain.  Outlines will be disabled.  To enable outlines, disable geometry terrain clamping by explicitly setting height to 0.` so that users actually know what they need to do to get the outline back if they want it.